components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/model/entity/API.java

```diff
@@ -375,4 +377,12 @@ public Map<String, String> getApiProperties() {
     public void setApiProperties(Map<String, String> apiProperties) {
         this.apiProperties = apiProperties;
     }
+
+    public List<VHost> getVhosts() {
+        return vhosts;
+    }
+
+    public void setVhosts(List<VHost> vhosts) {
+        this.vhosts = (vhosts != null) ? vhosts : new ArrayList<>();
```

Comment:
this can reset an already populated List.  Shall we do a null check and avoid setting anything if null? WDYT?

Reply:
yes, it can reset, fixed in https://github.com/wso2/carbon-apimgt/pull/13361/commits/ffa9706e9d639858e2886dad9b3788673018fcd9